sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala

```diff
@@ -766,6 +774,14 @@ class HiveThriftHttpServerSuite extends HiveThriftJdbcTest {
       assert(resultSet.getString(2) === HiveUtils.builtinHiveVersion)
     }
   }
+
+  test("Checks cast as float") {
```

Comment:
Duplicated code?

Reply:
for two different modes:  binary &  http

Reply:
then probably better to add it into HiveThriftJdbcTest?

Reply:
yes, so if you add the test to `HiveThriftJdbcTest` it will be executed in both in `HiveThriftBinaryServerSuite` and ` HiveThriftBinaryServerSuite`....

Reply:
sorry i do not agree with you. HiveThriftBinaryServerSuite and HiveThriftHttpServerSuite ,which have exist before, are different cases for different ServerMode, we should test them all. if we add test in HiveThriftJdbcTest , we need to set different ServerMode in HiveThriftJdbcTest for the two cases, which break the design of abstract class of HiveThriftJdbcTest .

Reply:
I don't think so. If you add:
```
  test("Checks cast as float") {
    withJdbcStatement() { statement =>
      val resultSet = statement.executeQuery("SELECT CAST('4.56' AS FLOAT)")
      resultSet.next()
      assert(resultSet.getString(1) === "4.56")
    }
  }
```
to the abstract class you need to change nothing else and the test would be executed both as a part of HiveThriftHttpServerSuite with one mode and as a part of HiveThriftBinaryServerSuite with the other mode.

Reply:
oh i see, i will fix it now. Thanks.